bridge-token-factory/src/lib.rs

```diff
@@ -263,12 +317,14 @@ impl BridgeTokenFactory {
             &self.prover_account,
             NO_DEPOSIT,
             env::prepaid_gas() / 4,
-        ).then(ext_self::finish_unlock(
+        )
+        .then(ext_self::finish_unlock(
             event.token,
             event.recipient,
             event.amount,
+            proof_1,
```

Comment:
Is it possible to do something like
```
ext_prover::verify_log_entry(
            proof.log_index,
            proof.log_entry_data,
            &self.prover_account,
            NO_DEPOSIT,
            env::prepaid_gas() / 4,
        ).then(record_proof(proof)).then(finish_unlock)
```
?

Reply:
@Kouprin we can do it, and probably a better way to do it is with:

```rust
ext_prover::verify_log_entry().then(finish_unlock).and(record_proof)
```

so both calls are resolved in parallel. But it is kind of weird to do it, since both functions belong to the same contract, and we must split gas between both, while we can just do it everything inside a single call.

Notice that the call to `record_proof` inside `finish_unlock` is not a cross-contract call, but just an inner call to one of the functions.